atlasprintServer/atlasprintService.py

```diff
@@ -0,0 +1,190 @@
+"""
+***************************************************************************
+    QGIS Server Plugin Filters: Add a new request to print a specific atlas
+    feature
+    ---------------------
+    Date                 : December 2019
+    Copyright            : (C) 2019 by René-Luc D'Hont - 3Liz
+    Email                : rldhont at 3liz dot com
+***************************************************************************
+*                                                                         *
+*   This program is free software; you can redistribute it and/or modify  *
+*   it under the terms of the GNU General Public License as published by  *
+*   the Free Software Foundation; either version 2 of the License, or     *
+*   (at your option) any later version.                                   *
+*                                                                         *
+***************************************************************************
+"""
+
+from pathlib import Path
+
+from qgis.server import (QgsService,
+                         QgsServerRequest,
+                         QgsServerResponse)
+
+from .core import print_atlas, AtlasPrintException
+
+
+def write_json_response( data: Dict[str,str], response: QgsServerResponse, code: int=200) -> None:
+    """ Write data as json response
+    """
+    response.setStatusCode(code)
+    response.setHeader("Content-Type", "application/json")
+    response.write(json.dumps(data))
+
+
+
+class AtlasPrintError(Exception):
+
+    def __init__(self, code: int, msg: str) -> None:
+        super().__init__(msg)
+        self.msg  = msg
+        self.code = code
+        QgsMessageLog.logMessage("Atlas print request error %s: %s" % (code,msg),"atlasprint",Qgis.Critical)
+
+    def formatResponse(self, response: QgsServerResponse) -> None:
+        """ Format error response
+        """
+        body = { 'status': 'fail', 'message': self.msg }
+        response.clear()
+        write_json_response(body, response, self.code)
```

Comment:
With @dmarteau we were saying that if the code is 500 (5XX), this should stay on the server side and not added in the body. We can fix that later.